src/hooks/character.js

```diff
@@ -0,0 +1,35 @@
+import { useEffect, useState } from 'react';
+import { getApi } from '../services/api.js';
+import { getApiByName } from '../services/api';
+
+export const useCharacters = () => {
```

Comment:
page needs to be a parameter for this hook - you pass it as an argument in your characterlist component but if you don't have it as a parameter in the function definition it won't know what to do with it